source/agora/test/EnrollmentManager.d

```diff
@@ -95,3 +95,135 @@ unittest
         retryFor(node.getPreimage(enroll.utxo_key) != PreImageInfo.init,
             10.seconds));
 }
+
+/// Situation: A node in the network crashes after enrolling as an validator.
+///     And the node restarts before a new block block containing the enrollment
+///     information is created. And the block is created, and the enrollment
+///     information of the new block is adding to the validator set.
+/// Expectation: The node restores its own enrollment data when the block
+///     containing the enrollment information arrives.
+unittest
+{
+    import agora.consensus.validation.PreImage;
+    import core.thread;
+    import std.conv;
+
+    auto network = makeTestNetwork(TestConf.init);
+    scope(exit) network.shutdown();
+    scope(failure) network.printLogs();
+    network.start();
+    network.waitForDiscovery();
+
+    // Boilerplate
+    auto nodes = network.clients;
+    auto gen_key_pair = getGenesisKeyPair();
+    auto node_1 = nodes[0];
+    auto pubkey_1 = node_1.getPublicKey();
+
+    // Create a new block with frozen transactions so we can Enroll
+    Transaction[] txs;
+    {
+        foreach (idx; 0 .. Block.TxsInBlock)
+        {
+            auto input = Input(hashFull(GenesisTransaction), idx.to!uint);
+            Transaction tx =
+                {
+                    TxType.Freeze,
+                    [input],
+                    [Output(Amount.MinFreezeAmount, pubkey_1),
+                        Output(Amount(1000), gen_key_pair.address)]
+                };
+            auto signature = gen_key_pair.secret.sign(hashFull(tx)[]);
+            tx.inputs[0].signature = signature;
+            txs ~= tx;
+        }
+        // Send all transactions to node 1 and rely on gossip
+        txs.each!(tx => node_1.putTransaction(tx));
+        // Make sure the block is created
+        containSameBlocks(nodes, 1).retryFor(5.seconds);
+    }
+
+    // The nodes[0] register as validator
+    auto enroll = node_1.createEnrollmentData();
+    node_1.enrollValidator(enroll);
+    nodes.each!(node =>
+        retryFor(node.getEnrollment(enroll.utxo_key) == enroll, 5.seconds));
+
+    // The `enroll` hasn't been validator yet so there must not be a pre-image
+    assert(node_1.getPreimage(enroll.utxo_key) == PreImageInfo.init);
+
+    // Check if the `node_1` restores block from the chain after restarting
+    node_1.restart();
+    RemoteAPI!TestAPI[] new_nodes;
```

Comment:
We don't need to create a new array, just use `nodes` as before.
That was the point of adding this feature to LocalRest directly instead of putting it here: If we needed a new client, then it would mean we'd have to handle this for *every running nodes*,  not only the main thread.